stats/opentelemetry/e2e_test.go

```diff
@@ -582,3 +601,365 @@ func pollForWantMetrics(ctx context.Context, t *testing.T, reader *metric.Manual
 
 	return fmt.Errorf("error waiting for metrics %v: %v", wantMetrics, ctx.Err())
 }
+
+// spanInformation is the information received about the span. This is a subset
+// of information that is important to verify that gRPC has knobs over, which
+// goes through a stable OpenTelemetry API with well-defined behavior. This keeps
+// the robustness of assertions over time.
+type spanInformation struct {
+	spanKind   string
+	name       string
+	events     []trace.Event
+	attributes []attribute.KeyValue
+}
+
+// TestClientCallSpanEvents verifies the events added to call spans
+// for a unary RPC, including events for gRPC status. It also verifies
+// the same traceID is propagated across client to server. It sets up a
+// stub server with OpenTelemetry tracing enabled, makes a unary RPC with
+// gzip compression, and then asserts that the exported spans contain the
+// expected events and attributes.
+func (s) TestClientCallSpanEvents(t *testing.T) {
```

Comment:
i think this should be just called TestSpan and it should test the span context on client and server for both unary and streaming case. Can you write test similar to this https://github.com/grpc/grpc-go/blob/4c07bca27377feb808912b844b3fa95ad10f946b/stats/opencensus/e2e_test.go#L1395